src/Microsoft.AspNetCore.Mvc.Core/Internal/AttributeRoute.cs

```diff
@@ -194,15 +208,16 @@ private static List<RouteInfo> GetRouteInfos(IReadOnlyList<ActionDescriptor> act
 
             try
             {
-                RouteTemplate parsedTemplate;
-                if (!templateCache.TryGetValue(action.AttributeRouteInfo.Template, out parsedTemplate))
+                if (!templateCache.TryGetValue(action.AttributeRouteInfo.Template, out var parsedTemplate))
                 {
                     // Parsing with throw if the template is invalid.
                     parsedTemplate = TemplateParser.Parse(action.AttributeRouteInfo.Template);
                     templateCache.Add(action.AttributeRouteInfo.Template, parsedTemplate);
                 }
 
                 routeInfo.RouteTemplate = parsedTemplate;
+                routeInfo.MapInbound = !action.AttributeRouteInfo.SuppressForPathMatching;
+                routeInfo.MapOutbound = !action.AttributeRouteInfo.SuppressForLinkGeneration;
```

Comment:
Just a personal preference thing, but I find this sort of thing really confusing. It would be more straightforward to just use the same property names and not invert anything.